packages/core/src/components/overlay/overlay.tsx

```diff
@@ -100,7 +100,7 @@ export class Overlay extends AbstractPureComponent<OverlayProps, OverlayState> {
         // TransitionGroup types require single array of children; does not support nested arrays.
         // So we must collapse backdrop and children into one array, and every item must be wrapped in a
         // Transition element (no ReactText allowed).
-        const childrenWithTransitions = isOpen ? Children.map(children, this.maybeRenderChild) ?? [] : [];
+        const childrenWithTransitions = isOpen ? (Children.map(children, this.maybeRenderChild) ?? []) : [];
```

Comment:
Prettier `3.3.3` added parentheses for nullish coalescing in ternaries for operator precedence clarity

https://github.com/prettier/prettier/blob/main/CHANGELOG.md#add-parentheses-for-nullish-coalescing-in-ternary-16391-by-cdignam-segment